File: src/main/java/fr/ynerant/leveleditor/api/editor/RawCase.java
package fr.ynerant.leveleditor.api.editor;

public class RawCase {
    private int x;
    private int y;
    private RawSprite couche1;
    private RawSprite couche2;
    private RawSprite couche3;
    private Collision collision;

    public static RawCase create(int posX, int posY, RawSprite couche1, RawSprite couche2, RawSprite couche3, Collision collision) {
        RawCase c = new RawCase();
        c.x = posX;
        c.y = posY;
        c.couche1 = couche1;
        c.couche2 = couche2;
        c.couche3 = couche3;
        c.collision = collision;
        return c;
    }

    public static RawCase create(Case c) {
        RawCase raw = new RawCase();
        raw.x = c.getPosX();
        raw.y = c.getPosY();
        raw.couche1 = RawSprite.create(c.getCoucheOne());
        raw.couche2 = RawSprite.create(c.getCoucheTwo());
        raw.couche3 = RawSprite.create(c.getCoucheThree());
        raw.collision = c.getCollision();
        return raw;
    }

    public int getPosX() {
        return x;
    }

    public int getPosY() {
        return y;
    }

    public RawSprite getCoucheOne() {
        return couche1;
    }

    public RawSprite getCoucheTwo() {
        return couche2;
    }

    public RawSprite getCoucheThree() {
        return couche3;
    }

    public Collision getCollision() {
        return collision;
    }
}

File: src/main/scala/fr/ynerant/leveleditor/api/editor/RawCase.scala
package fr.ynerant.leveleditor.api.editor

import fr.ynerant.leveleditor.api.editor.Collision.Collision

object RawCase {
	def create(posX: Int, posY: Int, couche1: RawSprite, couche2: RawSprite, couche3: RawSprite, collision: Collision): RawCase = {
		new RawCase(posX, posY, couche1, couche2, couche3, collision)
	}

	def create(c: Case): RawCase = {
		new RawCase(c.getPosX, c.getPosY, RawSprite.create(c.getCoucheOne), RawSprite.create(c.getCoucheTwo), RawSprite.create(c.getCoucheThree), c.getCollision)
	}
}

case class RawCase(var x: Int, var y: Int, var couche1: RawSprite, var couche2: RawSprite, var couche3: RawSprite, var collision: Collision) {
	def getPosX: Int = x

	def getPosY: Int = y

	def getCoucheOne: RawSprite = couche1

	def getCoucheTwo: RawSprite = couche2

	def getCoucheThree: RawSprite = couche3

	def getCollision: Collision = collision

	def setCollision(collision: Collision): Unit = {
		this.collision = collision
	}
}
